x-pack/plugins/security_solution/common/detection_engine/get_query_filter.ts

```diff
@@ -31,24 +32,93 @@ export const getQueryFilter = (
+    queryStringOptions: { analyze_wildcard: true },
+    ignoreFilterIfFieldNotInIndex: false,
+    dateFormatTZ: 'Zulu',
+  };
+
+  const enabledFilters = ((filters as unknown) as Filter[]).filter((f) => !isFilterDisabled(f));
   /*
    * Pinning exceptions to 'kuery' because lucene
    * does not support nested queries, while our exceptions
    * UI does, since we can pass both lucene and kql into
    * buildEsQuery, this allows us to offer nested queries
    * regardless
    */
-  const exceptions = buildQueryExceptions({ language: 'kuery', lists, exclude: excludeExceptions });
-  const queries: DataQuery[] = [...initialQuery, ...exceptions];
+  const exceptionQueries = buildExceptionListQueries({ language: 'kuery', lists });
+  if (exceptionQueries.length > 0) {
+    // Assume that `indices.query.bool.max_clause_count` is at least 1024 (the default value),
+    // allowing us to make 1024-item chunks of exception list items.
+    // Discussion at https://issues.apache.org/jira/browse/LUCENE-4835 indicates that 1024 is a
+    // very conservative value.
+    const exceptionFilter = buildExceptionFilter(
+      exceptionQueries,
+      indexPattern,
+      config,
+      excludeExceptions,
+      1024
+    );
+    enabledFilters.push(exceptionFilter);
+  }
+  const initialQuery = { query, language };
 
-  const config = {
-    allowLeadingWildcards: true,
-    queryStringOptions: { analyze_wildcard: true },
-    ignoreFilterIfFieldNotInIndex: false,
-    dateFormatTZ: 'Zulu',
-  };
+  return buildEsQuery(indexPattern, initialQuery, enabledFilters, config);
+};
 
-  const enabledFilters = ((filters as unknown) as Filter[]).filter((f) => !isFilterDisabled(f));
-  return buildEsQuery(indexPattern, queries, enabledFilters, config);
+export const buildExceptionFilter = (
+  exceptionQueries: Query[],
+  indexPattern: IIndexPattern,
+  config: EsQueryConfig,
+  excludeExceptions: boolean,
+  chunkSize: number
+) => {
+  const exceptionFilter: Filter = {
+    meta: {
+      alias: null,
+      negate: excludeExceptions,
+      disabled: false,
+    },
+    query: {
+      bool: {
+        should: undefined,
+      },
+    },
+  };
+  if (exceptionQueries.length <= chunkSize) {
+    const query = buildEsQuery(indexPattern, exceptionQueries, [], config);
+    exceptionFilter.query.bool.should = query.bool.filter;
+  } else {
+    const chunkedFilters: Filter[] = [];
+    for (let index = 0; index < exceptionQueries.length; index += chunkSize) {
+      const exceptionQueriesChunk = exceptionQueries.slice(index, index + chunkSize);
+      const esQueryChunk = buildEsQuery(indexPattern, exceptionQueriesChunk, [], config);
+      const filterChunk: Filter = {
+        meta: {
+          alias: null,
+          negate: false,
+          disabled: false,
+        },
+        query: {
+          bool: {
+            should: esQueryChunk.bool.filter,
+          },
+        },
+      };
+      chunkedFilters.push(filterChunk);
+    }
```

Comment:
Is it possible to rewrite this with a `map`? Tendency to prefer a functional style for situations like this.